modin/core/io/sql/sql_dispatcher.py

```diff
@@ -56,34 +57,24 @@ def _read(cls, sql, con, index_col=None, **kwargs):
         BaseQueryCompiler
             Query compiler with imported data for further processing.
         """
-        try:
-            import psycopg2 as pg
-
-            if isinstance(con, pg.extensions.connection):
-                con = "postgresql+psycopg2://{}:{}@{}{}/{}".format(  # Table in DB
-                    con.info.user,  # <Username>: for DB
-                    con.info.password,  # Password for DB
-                    con.info.host if con.info.host != "/tmp" else "",  # @<Hostname>
-                    (":" + str(con.info.port))
-                    if con.info.host != "/tmp"
-                    else "",  # <port>
-                    con.info.dbname,  # Table in DB
-                )
-        except ImportError:
-            pass
-        # In the case that we are given a SQLAlchemy Connection or Engine, the objects
-        # are not pickleable. We have to convert it to the URL string and connect from
-        # each of the workers.
-        if not isinstance(con, str):
+        connection_is_modin_db_connection = isinstance(con, ModinDatabaseConnection)
+        if not (connection_is_modin_db_connection or isinstance(con, str)):
             warnings.warn(
-                "To use parallel implementation of `read_sql`, pass the sqlalchemy"
-                "connection string instead of {}.".format(type(con))
+                "To use parallel implementation of `read_sql`, pass either "
+                "the SQL connection string or a ModinDatabaseConnection "
```

Comment:
Ideally here we'd point to a webpage where we explain how to use `ModinDatabaseConnection`.

Reply:
Done.